getting nil pointer dereference on `thyme show -w stats` (macOS)

I'm using GO version 1.7, and I got SIGSEGV when trying to exectute `thyme show -i thyme.json -w stats > thyme.html`. I'm not sure if 1.7 is causing the issue though.

Thanks!

Here are the details:

```
applefreak:~/ $ thyme show -i thyme.json -w stats > thyme.html                                     [11:34:25]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xa9f95]

goroutine 1 [running]:
panic(0x17f600, 0xc420010070)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/sourcegraph/thyme.Stats.func1(0x0, 0xc4202ed860, 0x9cb)
	/Users/applefreak/go/src/github.com/sourcegraph/thyme/show.go:17 +0x5
github.com/sourcegraph/thyme.NewTimeline(0xc4200a8420, 0x1c1c28, 0xc4202edc48)
	/Users/applefreak/go/src/github.com/sourcegraph/thyme/show.go:142 +0x39e
github.com/sourcegraph/thyme.Stats(0xc4200a8420, 0x5, 0x1aa813)
	/Users/applefreak/go/src/github.com/sourcegraph/thyme/show.go:17 +0x3b
main.(*ShowCmd).Execute(0x278180, 0xc4200ba640, 0x0, 0x5, 0x0, 0x0)
	/Users/applefreak/go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:114 +0x44b
github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc4200d6000, 0xc42000a130, 0x5, 0x5, 0x1, 0x0, 0x1b3201, 0xc4200001a0, 0x0)
	/Users/applefreak/go/src/github.com/jessevdk/go-flags/parser.go:314 +0x8e6
github.com/jessevdk/go-flags.(*Parser).Parse(0xc4200d6000, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/applefreak/go/src/github.com/jessevdk/go-flags/parser.go:185 +0x74
main.main.func1(0x0, 0x32c4)
	/Users/applefreak/go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:141 +0x2d
main.main()
	/Users/applefreak/go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:148 +0x31
```

```thyme show -i thyme.json``` outputs a text file, but with ```-w stats``` it errors out.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4bd9e0]

goroutine 1 [running]:
panic(0x5f69e0, 0xc82000e0f0)
	/usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
github.com/sourcegraph/thyme.Stats.func1(0x0, 0xc820069618, 0x1c770c8)
	/home/millette/gocode/src/github.com/sourcegraph/thyme/show.go:17 +0x20
github.com/sourcegraph/thyme.NewTimeline(0xc8200ae540, 0x69a658, 0xc8200ae540)
	/home/millette/gocode/src/github.com/sourcegraph/thyme/show.go:142 +0x3d8
github.com/sourcegraph/thyme.Stats(0xc8200ae540, 0x0, 0x0)
	/home/millette/gocode/src/github.com/sourcegraph/thyme/show.go:17 +0x46
main.(*ShowCmd).Execute(0x758400, 0xc8200b4730, 0x0, 0x5, 0x0, 0x0)
	/home/millette/gocode/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:114 +0x59e
github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc8200d0000, 0xc82000a0d0, 0x5, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/millette/gocode/src/github.com/jessevdk/go-flags/parser.go:314 +0xa83
github.com/jessevdk/go-flags.(*Parser).Parse(0xc8200d0000, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/millette/gocode/src/github.com/jessevdk/go-flags/parser.go:185 +0x9b
main.main.func1(0x0, 0x0)
	/home/millette/gocode/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:141 +0x2f
main.main()
	/home/millette/gocode/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:148 +0x23
```

But I'm on Debian GNU/Linux. It worked a few times but it won't generate html anymore.

Yep, I have the same issue.

```bash
┬─[william@fillory:~/Projects/thyme/data]─[04:33:10 PM]
╰─>$ thyme show -i 20160817.json -w stats
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4bd9e0]

goroutine 1 [running]:
panic(0x5f69e0, 0xc820012100)
	/usr/lib/go/src/runtime/panic.go:481 +0x3e6
github.com/sourcegraph/thyme.Stats.func1(0x0, 0xc8200a7608, 0x1a0155e)
	/home/william/go/src/github.com/sourcegraph/thyme/show.go:17 +0x20
github.com/sourcegraph/thyme.NewTimeline(0xc8200b03e0, 0x69a658, 0xc8200b03e0)
	/home/william/go/src/github.com/sourcegraph/thyme/show.go:142 +0x3d8
github.com/sourcegraph/thyme.Stats(0xc8200b03e0, 0x0, 0x0)
	/home/william/go/src/github.com/sourcegraph/thyme/show.go:17 +0x46
main.(*ShowCmd).Execute(0x758400, 0xc8200b2460, 0x0, 0x5, 0x0, 0x0)
	/home/william/go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:114 +0x59e
github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc8200d2000, 0xc82000a130, 0x5, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/william/go/src/github.com/jessevdk/go-flags/parser.go:314 +0xa83
github.com/jessevdk/go-flags.(*Parser).Parse(0xc8200d2000, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/william/go/src/github.com/jessevdk/go-flags/parser.go:185 +0x9b
main.main.func1(0x0, 0x0)
	/home/william/go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:141 +0x2f
main.main()
	/home/william/go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:148 +0x23
```

same issue on windows:

```
$ thyme show -i thyme.json -w stats > thyme.html
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x4b3a35]

goroutine 1 [running]:
panic(0x59eb60, 0xc0420040c0)
        C:/tools/Go/src/runtime/panic.go:500 +0x1af
github.com/sourcegraph/thyme.Stats.func1(0x0, 0xc04202d858, 0x0)
        C:/tools/Go/src/github.com/sourcegraph/thyme/show.go:17 +0x5
github.com/sourcegraph/thyme.NewTimeline(0xc04200aee0, 0x5e4df8, 0xc04202dc40)
        C:/tools/Go/src/github.com/sourcegraph/thyme/show.go:142 +0x3a5
github.com/sourcegraph/thyme.Stats(0xc04200aee0, 0x5, 0x5cb2f8)
        C:/tools/Go/src/github.com/sourcegraph/thyme/show.go:17 +0x42
main.(*ShowCmd).Execute(0x6a3680, 0xc04200e690, 0x0, 0x5, 0x0, 0x0)
        C:/tools/Go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:114 +0x452
github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc0420104e0, 0xc042010250, 0x5, 0x5, 0x1, 0x0, 0x5d5101, 0xc04201a000, 0x0)
        C:/tools/Go/src/github.com/jessevdk/go-flags/parser.go:314 +0x8e3
github.com/jessevdk/go-flags.(*Parser).Parse(0xc0420104e0, 0x0, 0x0, 0x0, 0x0, 0x0)
        C:/tools/Go/src/github.com/jessevdk/go-flags/parser.go:185 +0x7b
main.main.func1(0x0, 0x40230b)
        C:/tools/Go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:141 +0x34
main.main()
        C:/tools/Go/src/github.com/sourcegraph/thyme/cmd/thyme/main.go:148 +0x38
```

Go version: ``` go version go1.7 windows/amd64 ```

--- a/show.go
+++ b/show.go
@@ -44,7 +44,9 @@ func NewAggTime(stream *Stream, labelFunc func(*Window) string) *AggTime {
 			windows[win.ID] = win
 		}
 
-		active.Plus(labelFunc(windows[snap.Active]), 1)
+		if win := windows[snap.Active]; win != nil {
+			active.Plus(labelFunc(windows[snap.Active]), 1)
+		}
 		for _, v := range snap.Visible {
 			visible.Plus(labelFunc(windows[v]), 1)
 		}
@@ -138,17 +140,20 @@ func NewTimeline(stream *Stream, labelFunc func(*Window) string) *Timeline {
 		}
 
 		{
-			win := windows[snap.Active]
-			winLabel := labelFunc(win)
-			if lastActive != nil && lastActive.Label == winLabel {
-				lastActive.End = snap.Time
-			} else {
-				if lastActive != nil {
+			if win := windows[snap.Active]; win != nil {
+				winLabel := labelFunc(win)
+				if lastActive != nil && lastActive.Label == winLabel {
 					lastActive.End = snap.Time
+				} else {
+					if lastActive != nil {
+						lastActive.End = snap.Time
+					}
+					newRange := &Range{Label: winLabel, Start: snap.Time, End: snap.Time}
+					active = append(active, newRange)
+					lastActive = newRange
 				}
-				newRange := &Range{Label: winLabel, Start: snap.Time, End: snap.Time}
-				active = append(active, newRange)
-				lastActive = newRange
+			} else {
+				lastActive = nil
 			}
 		}
 

